src/Grpc.Net.Client/GrpcChannel.cs

```diff
@@ -221,7 +369,22 @@ private GrpcMethodInfo CreateMethodInfo(IMethod method)
             var scope = new GrpcCallScope(method.Type, uri);
             var methodConfig = ResolveMethodConfig(method);
 
-            return new GrpcMethodInfo(scope, new Uri(Address, uri), methodConfig);
+            var uriBuilder = new UriBuilder(Address);
+            uriBuilder.Path = method.FullName;
+
+            // The Uri used to create HttpRequestMessage must have a http or https scheme.
+            uriBuilder.Scheme = IsSecure ? Uri.UriSchemeHttps : Uri.UriSchemeHttp;
+
+            // A Uri with a http or https scheme requires a host name.
+            // Triple slash URIs, e.g. dns:///custom-value, won't have a host and UriBuilder throws an error.
+            // Add a temp value as the host. The tempuri.org host may show up in some logging but it will
+            // get replaced in the final HTTP request address by the load balancer.
+            if (string.IsNullOrEmpty(uriBuilder.Host))
+            {
+                uriBuilder.Host = "tempuri.org";
+            }
```

Comment:
you should at least use a hostname that is guaranteed to be invalid.

https://datatracker.ietf.org/doc/html/rfc2606#section-2

> .invalid" is intended for use in online construction of domain
      names that are sure to be invalid and which it is obvious at a
      glance are invalid.

so using e.g. "loadbalancer.determines.real.hostname.invalid" seems much less cryptic when it shows up somewhere.

Reply:
Changed to `loadbalancer.temporary.invalid`